### DISCUSSION TITLE
Versions.

### DISCUSSION THREAD

**Discussion by FlatrockITSolutions** [avendesora/pythonbible#73] [General]
Awesome Resource. Do you have more versions of bibles to chose from? (Aside from NKJ and ASV which seems to be a carbon copy of NKJ) If not, what is the formatting for the data so I can compile my own versions, and upload them. I am working on building my own Discord bot that has conversations with users, and thought to add features like understanding a scripture reference whenever it is sent in chat. Here is the source code for my discord bot. https://github.com/FlatrockDesigns/The-Kevin-Project I am primarily looking for NKJV and AMPLIFIED. I found Code for the versions but I did not find source files for them. I even did pip install pythonbible[all] with no luck.

**Comment by avendesora** [comment]
Great question! 

1. Yes, the intention is definitely that the pythonbible project will support any version of the Bible. However...

2. Due to copyright, we are only planning on including the Public Domain (or otherwise copyright-free) versions of the Bible in the actual project source code.

Currently, there is a parser library in the pythonbible project. I'm in the process of separating that out into its own [project](https://github.com/avendesora/pythonbible-parser). That can then be used to take the source file(s) for any version of the Bible (as long as they are in a supported format) and generate the python files necessary to use that version with the pythonbible project. So, to use the New King James and Amplified versions you would need to license them from the copyright holder (currently Harper Collins and Zondervan, respectively), or some allow for free use in non-profit scenarios, obtain the source files for those versions, and then run them through the pythonbible-parser for use in the pythonbible project.

Some of this is still a work in progress (hopefully completed by the end of the year, but no promises/guarantees). If you can get the necessary license and source files, I would be happy to work with you to get those versions supported (though not included as part of the source code).

Thanks for the question! I love to hear how others are using this project.

**Reply by FlatrockITSolutions** [reply]
Since it is opensource hence none profit of sorts I assumed that copywrite laws would not be a factor. Would there be a work around for citing the source/ copy write owner to give credit where credit is due to allow it to be used in an opensource/none-profit format? I am new to copywrite infringement laws as you can probably tell from the above question.

**Reply by avendesora** [reply]
You should check out the permissions pages for both Thomas Nelson (for NKJV) and Zondervan (for Amplified), though both appear to be owned by HarperCollins now, so the process would probably be the same:

[https://www.thomasnelson.com/about-us/permissions/](https://www.thomasnelson.com/about-us/permissions/)
[https://www.zondervan.com/about-us/permissions/](https://www.zondervan.com/about-us/permissions/)

Copyright is definitely a factor regardless of open source, maybe, especially because of open source. We can't include copyrighted versions in the source code of the pythonbible library, because we don't restrict how the pythonbible library can be used. It can be used for both commercial and non-commercial purposes. But if you get permission (and the source data and run it through the pythonbible-parser to generate the necessary files), you can use those versions with the pythonbible library. (Some work might be necessary on both the pythonbible-parser and pythonbible for that, but I would be happy to help.)

I would imagine they would be more likely to grant you permission for a closed-source project rather than an open-source project since I would assume they would not want it publicly available on GitHub, but that's just an assumption on my part.

Standard Disclaimer: I am not a copyright lawyer, and you should not treat this as legal advice. But I would be very hesitant to use copyrighted material without either getting permission from the copyright holder and/or discussing it with an actual copyright lawyer.
